src/datepicker/calendar-header.js

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+import * as React from 'react';
+import {ArrowLeft, ArrowRight} from '../icon/index.js';
+import {Select} from '../select/index.js';
+import {StyledHeader} from './styled-components.js';
+import {
+  addMonths,
+  getMonth,
+  getMonthInLocale,
+  getYear,
+  monthDisabledBefore,
+  monthDisabledAfter,
+  setMonth,
+  setYear,
+  subMonths,
+} from './utils/index.js';
+import type {HeaderPropsT} from './types.js';
+
+export default class CalendarHeader extends React.Component<HeaderPropsT> {
+  static defaultProps = {
+    date: new Date(),
+    locale: null,
+    maxDate: null,
+    minDate: null,
+    onMonthChange: () => {},
+    onYearChange: () => {},
+    setActiveState: () => {},
+  };
+
+  handleMonthChange = ({value}: {value: Array<{id: number}>}) => {
+    this.props.onMonthChange({date: setMonth(this.props.date, value[0].id)});
+  };
+
+  handleYearChange = ({value}: {value: Array<{id: number}>}) => {
+    this.props.onYearChange({date: setYear(this.props.date, value[0].id)});
+  };
+
+  increaseMonth = () => {
+    this.props.onMonthChange({date: addMonths(this.props.date, 1)});
+  };
+
+  decreaseMonth = () => {
+    this.props.onMonthChange({date: subMonths(this.props.date, 1)});
+  };
+
+  renderPreviousMonthButton = () => {
+    const allPrevDaysDisabled = monthDisabledBefore(
+      this.props.date,
+      this.props,
+    );
+    if (allPrevDaysDisabled) {
+      return;
+    }
+    let clickHandler = this.decreaseMonth;
+    if (allPrevDaysDisabled) {
+      clickHandler = null;
+    }
+    return (
+      <ArrowLeft
+        role="button"
+        onClick={clickHandler}
+        overrides={{Svg: {style: {cursor: 'pointer'}}}}
+      />
+    );
+  };
+
+  renderNextMonthButton = () => {
+    const allNextDaysDisabled = monthDisabledAfter(this.props.date, this.props);
+    if (allNextDaysDisabled) {
+      return;
+    }
+    let clickHandler = this.increaseMonth;
+    if (allNextDaysDisabled) {
+      clickHandler = null;
+    }
+    return (
+      <ArrowRight
+        role="button"
+        onClick={clickHandler}
+        overrides={{Svg: {style: {cursor: 'pointer'}}}}
+      />
+    );
+  };
+
+  getSelectOverrides({width}: {width: string}) {
+    return {
+      ControlContainer: {
+        style: props => {
+          const {
+            $isFocused,
+            $isPseudoFocused,
+            $theme: {colors},
+          } = props;
+          return {
+            width,
+            backgroundColor: colors.primary,
+            color: colors.white,
+            borderColor:
+              $isFocused || $isPseudoFocused ? colors.white : colors.primary,
```

Comment:
What do you think about using a dark background color for these things instead of a white border?

| before  | after  |
|---|---|
| ![image](https://user-images.githubusercontent.com/875591/50122721-bc1a0700-0212-11e9-84c0-bf5949f812dc.png)  | ![image](https://user-images.githubusercontent.com/875591/50122690-9ab91b00-0212-11e9-9e6e-e5cfae2a46fe.png)  |

Reply:
cc @nguyensomniac

Reply:
Dark background looks really good. Will change it.